server/aesdsocket.c

```diff
@@ -161,513 +167,497 @@ int main(int argc, char* argv[])
-				printf("recv() succeeded\n");
-			#endif
-		}
+		/*------------------------------------------------------------------------*/
 		for(i = 0;i < BUFFER_SIZE;i++)
-		{ // check for end of packet
+		{
 			if(read_data[i] == '\n')
 			{
-				packet_rcvd = true;		//end of packet detected, while loop should be 
-				i++;					//exited for further packet processing
+				packet_rcvd = true;
+				i++;
 				break;
 			}
 		}
-		packets += i;			//update the size of packet
-		buffer = (char *) realloc(buffer,(size_t)(packets+1));
+		packets += i;
+		/*------------------------------------------------------------------------*/
+		buffer = (char *) realloc(buffer,(ssize_t)(packets+1));
 		if(buffer == NULL)
 		{
-			syslog(LOG_ERR, "Realloc failed!\n");
+			syslog(LOG_ERR,"realloc() failed");
 			#if DEBUG
-				printf("realloc() failed\n");
+				printf("recv() failed");
 			#endif
-			clear();
 			exit(EXIT_FAILURE);
 		}
-		syslog(LOG_INFO, "realloc() succeeded!\n");
-		#if DEBUG
-			printf("realloc() succeeded\n");
-		#endif
 		strncat(buffer,read_data,i);
 		memset(read_data,0,BUFFER_SIZE);
 	}
-
+	/*------------------------------------------------------------------------*/
 	status = pthread_mutex_lock(parameters->mutex);
-	if (status != SUCCESS)
+	if(status != SUCCESS)
 	{
-		syslog(LOG_ERR, "pthread_mutex_lock() failed with error number %d\n", status);
-		clear();
+		syslog(LOG_ERR,"pthread_mutex_lock() failed with error code: %d\n",status);
+		#if DEBUG
+			printf("pthread_mutex_lock() failed with error code: %d\n",status);
+		#endif
 		exit(EXIT_FAILURE);
 	}
-
 	/*------------------------------------------------------------------------*/
-	fd = open(STORAGE_PATH, O_APPEND | O_WRONLY);		//open the file to write the received data
+	fd = open(STORAGE_PATH,O_APPEND | O_WRONLY);
 	if(fd == ERROR)
 	{
-		syslog(LOG_ERR, "File open failed (append, writeonly) \n");
+		syslog(LOG_ERR,"open() failed (append and write only)\n");
 		#if DEBUG
-			printf("open() (append, writeonly) failed\n");
+			printf("open() failed (append and write only)\n");
 		#endif
-		clear();
 		exit(EXIT_FAILURE);
 	}
-	syslog(LOG_INFO, "open() succeeded!\n");
-	#if DEBUG
-		printf("open() (append, writeonly) succeeded\n");
-	#endif
 	/*------------------------------------------------------------------------*/
-	nwrite = write(fd, buffer, strlen(buffer));	//start writing to the file
-	if (nwrite == ERROR)
+	int status = write(fd,buffer,strlen(buffer));
+	if(status == ERROR)
 	{
-		syslog(LOG_ERR, "write() failed\n");
+		syslog(LOG_ERR,"write() failed\n");
 		#if DEBUG
 			printf("write() failed\n");
 		#endif
-		clear();
 		exit(EXIT_FAILURE);
 	}
-	else if (nwrite != strlen(buffer))
+	else if(status != strlen(buffer))
 	{
-		syslog(LOG_ERR, "file partially written\n");
+		syslog(LOG_ERR,"File partially written\n");
 		#if DEBUG
-			printf("file partially written\n");
+			printf("File partially written\n");
 		#endif
-		clear();
-		exit(EXIT_FAILURE);	
+		exit(EXIT_FAILURE);
```

Comment:
is exit the right way to clean up? what happens when you cancel program  to the pthrrads created?